src/Migrations/M036MigrateCoversContentBlockToPostsListBlock.php

```diff
@@ -0,0 +1,368 @@
+                                    [
+                                        Utils\Functions::create_new_block(
+                                            Utils\Constants::BLOCK_TERMS,
+                                            ['term' => 'post_tag', 'separator' => ' ']
+                                        ),
+                                        Utils\Functions::create_new_block(
+                                            Utils\Constants::BLOCK_TERMS,
+                                            ['term' => 'category', 'separator' => ' | ']
+                                        ),
+                                    ],
+                                    ['className' => 'posts-list-meta']
+                                ),
+                                Utils\Functions::create_new_block(
+                                    Utils\Constants::BLOCK_TITLE
+                                ),
+                                Utils\Functions::create_new_block(
+                                    Utils\Constants::BLOCK_EXCERPT
+                                ),
+                                Utils\Functions::create_group_block(
+                                    [
+                                        Utils\Functions::create_new_block(
+                                            Utils\Constants::BLOCK_AUTHOR,
+                                            ['isLink' => true]
+                                        ),
+                                        Utils\Functions::create_new_block(
+                                            Utils\Constants::BLOCK_DATE
+                                        ),
+                                    ],
+                                    ['className' => 'posts-list-meta',]
+                                ),
+                            ],
+                            []
+                        ),
+                    ]
+                ),
+            ],
+            [
+                'lock' => [
+                    'move' => true,
+                    'remove' => true,
+                ],
+            ]
+        );
+    }
+
+    /**
+     * Create and get a new navigation link block.
+     *
+     * @return mixed - The navigation link block if the All Stories page is set.
+     */
+    private static function get_nav_links_block(): array
+    {
+        if (!get_option('page_for_posts')) {
+            return [];
+        }
+
+        return Utils\Functions::create_new_block(
+            Utils\Constants::BLOCK_NAV_LINK,
+            [
+                'label' => __('See all stories', 'planet4-blocks'),
+                'url' => get_permalink(get_option('page_for_posts')),
+                'className' => 'see-all-link',
+            ]
+        );
+    }
+
+    /**
+     * Create and get a new query-no-results block.
+     *
+     * @return array - The new block.
+     */
+    private static function get_query_no_results_block(): array
```

Comment:
I'm wondering if we really need this 🤔 Do we want to migrate blocks that show 0 covers?
If we do then I would just use `No results found` as value, so that it shows properly in the frontend

Reply:
Hm.. I think we should keep them. Since the block is rendered dynamically based on the editor's choices. So it may be empty now, but it can have some Posts matching the editor's filter criteria at a later point.

Reply:
@mleray now the migration script shows in the frontend the "No posts found" message in empty Post List blocks.